blockchain/transaction.go

```diff
@@ -0,0 +1,71 @@
+package blockchain
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+)
+
+// TxHashPointer is a reference to a transaction on the blockchain.
+type TxHashPointer struct {
+	blockNumber uint32
+	hash        Hash
+}
+
+// Marshal converts a TxHashPointer to a byte slice
+func (thp TxHashPointer) Marshal() []byte {
+	buf := []byte{}
+	binary.LittleEndian.PutUint32(buf, thp.blockNumber)
+	for _, b := range thp.hash {
+		buf = append(buf, b)
+	}
+	return buf
+}
+
+// TxOutput defines an output to a transaction
+type TxOutput struct {
+	amount    uint64
+	recipient Wallet
+}
+
+// Marshal converts a TxOutput to a byte slice
+func (to TxOutput) Marshal() []byte {
+	buf := []byte{}
+	binary.LittleEndian.PutUint64(buf, to.amount)
+	buf = append(buf, to.recipient.Marshal()...)
+	return buf
+}
+
+// TxBody contains all relevant information about a transaction
+type TxBody struct {
+	sender  Wallet
+	input   TxHashPointer
+	outputs []TxOutput
```

Comment:
I think a transaction is unique because the `input` must always be unique, but I'm not yet sure 🤔